packages/core/src/components/callout/callout.tsx

```diff
@@ -26,18 +30,36 @@ export interface ICalloutProps extends IIntentProps, IProps {
 
 export class Callout extends React.PureComponent<ICalloutProps & React.HTMLAttributes<HTMLDivElement>, {}> {
     public render() {
-        const { className, children, iconName, intent, title, ...htmlProps } = this.props;
-        const classes = classNames(
-            Classes.CALLOUT,
-            Classes.intentClass(intent),
-            Classes.iconClass(iconName),
-            className,
-        );
+        const { className, children, iconName: _nospread, intent, title, ...htmlProps } = this.props;
+        const iconName = this.getIconName();
+        const classes = classNames(Classes.CALLOUT, Classes.intentClass(intent), className);
         return (
             <div className={classes} {...htmlProps}>
+                {iconName && (
+                    <span className={Classes.CALLOUT_ICON}>
+                        <Icon iconName={iconName} iconSize={Icon.SIZE_LARGE} />
+                    </span>
+                )}
                 {title && <h5>{title}</h5>}
                 {children}
             </div>
         );
     }
+
+    private getIconName(): IconName {
+        const { iconName, intent } = this.props;
+        if (iconName != null || intent === Intent.NONE) {
+            return iconName;
+        }
+        switch (intent) {
+            case Intent.DANGER:
+                return "error";
+            case Intent.PRIMARY:
+                return "info-sign";
+            case Intent.WARNING:
+                return "warning-sign";
+            case Intent.SUCCESS:
+                return "tick";
+        }
```

Comment:
would be nice to have an explicit `return undefined;` at the end rather than an implicit return.